File: Smartify_Mobil/app/src/main/java/com/example/smartify/api/models/Order.kt
package com.example.smartify.api.models

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Order(
    @SerializedName("_id")
    val id: String,
    val user: String,
    @SerializedName("orderItems")
    val items: List<OrderItem>,
    val shippingAddress: ShippingAddress,
    val paymentMethod: String,
    val totalPrice: Double,
    val shippingPrice: Double,
    val taxPrice: Double,
    val isPaid: Boolean,
    val paidAt: String? = null,
    val isDelivered: Boolean,
    val status: String,
    val createdAt: String,
    val updatedAt: String
) : Parcelable

@Parcelize
data class OrderItem(
    val product: String,
    val name: String,
    val price: Double,
    val quantity: Int,
    val image: String
) : Parcelable

@Parcelize
data class ShippingAddress(
    val firstName: String,
    val lastName: String,
    val address: String,
    val city: String,
    val postalCode: String,
    val phone: String,
    val email: String
) : Parcelable 

File: Smartify_Web/backend/models/Order.js
const mongoose = require('mongoose');

const orderItemSchema = new mongoose.Schema({
    product: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Product',
        required: true
    },
    name: {
        type: String,
        required: true
    },
    price: {
        type: Number,
        required: true
    },
    quantity: {
        type: Number,
        required: true,
        min: 1
    },
    image: {
        type: String
    }
});

const orderSchema = new mongoose.Schema({
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    orderItems: [orderItemSchema],
    shippingAddress: {
        firstName: { type: String, required: true },
        lastName: { type: String, required: true },
        address: { type: String, required: true },
        city: { type: String, required: true },
        postalCode: { type: String, required: true },
        phone: { type: String, required: true },
        email: { type: String, required: true }
    },
    paymentMethod: {
        type: String,
        required: true,
        enum: ['Havale/EFT', 'Kapıda Ödeme', 'Kredi Kartı']
    },
    paymentResult: {
        id: { type: String },
        status: { type: String },
        updateTime: { type: String },
        emailAddress: { type: String }
    },
    totalPrice: {
        type: Number,
        required: true,
        default: 0.0
    },
    shippingPrice: {
        type: Number,
        required: true,
        default: 0.0
    },
    taxPrice: {
        type: Number,
        required: true,
        default: 0.0
    },
    isPaid: {
        type: Boolean,
        required: true,
        default: false
    },
    paidAt: {
        type: Date
    },
    isDelivered: {
        type: Boolean,
        required: true,
        default: false
    },
    deliveredAt: {
        type: Date
    },
    status: {
        type: String,
        required: true,
        default: 'Beklemede',
        enum: ['Beklemede', 'İşleme Alındı', 'Kargoya Verildi', 'Teslim Edildi', 'İptal Edildi']
    },
    notes: {
        type: String
    }
}, {
    timestamps: true
});

module.exports = mongoose.model('Order', orderSchema); 